storage/src/main/java/io/delta/storage/S3SingleDriverLogStore.java

```diff
@@ -208,6 +208,7 @@ private Iterator<FileStatus> listFromInternal(
         if (
             // LocalFileSystem and RawLocalFileSystem checks are needed for tests to pass
             fs instanceof LocalFileSystem || fs instanceof RawLocalFileSystem || !enableFastListFrom
+                    || !FileNameUtils.isDeltaFilePrefix(resolvedPath)
```

Comment:
> Instead of adding this, could you take a look at the new `fastS3AListFrom` method and make it not list subdirectories?

Thank you for your suggestion. At the beginning, I planned to use the non-recursive method of `fastS3AListFrom`, but I didn't find it. Now this method only uses `fastS3AListFrom` for the _delta_log directory, which also conforms to the design of [#1210](https://github.com/delta-io/delta/pull/1210).

Reply:
In addition, `fastS3AListFrom` is always prefix range traversal, the original `listStatus` is full-text traversal. The two are not equivalent. The default resolvedPath for data file traversal is "\u0000", which will not be a problem, but if it is unintentionally modified, it may will cause a bug that only reads part of the data file.

Reply:
I found a non-recursive way of `fastS3AListFrom`, I will modify a version and add unit test.
```
new ListObjectsV2Request()
.withBucketName(s3afs.getBucket())
.withMaxKeys(maxKeys)
.withPrefix(s3afs.pathToKey(parentPath) + "/")
.withDelimiter("/")
.withStartAfter(keyBefore(s3afs.pathToKey(resolvedPath)))```